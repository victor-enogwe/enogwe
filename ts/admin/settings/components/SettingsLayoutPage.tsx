import { Fragment } from '@wordpress/element'
import * as React from 'react'
/**
 * SettingsLayoutPage Component
 *
 * @export
 * @returns {JSX.Element}
 */
export function SettingsLayoutPage(): JSX.Element {
    return (<Fragment />)
}
