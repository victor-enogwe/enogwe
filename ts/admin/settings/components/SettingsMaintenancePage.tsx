import { Fragment } from '@types/@wordpress/element'
import * as React from 'react'

export const SettingsMaintenancePage = () => {
    //   const settings = useContext(StateContext).maintenance

  return (<Fragment />)
}
