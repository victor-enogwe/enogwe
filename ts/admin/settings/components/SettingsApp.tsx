import * as React from 'react'
import { useEffect, createContext, useState } from '@wordpress/element'
import { Settings } from './Settings'
import { settingsStore } from '../store'
import { State } from '../store/state'
import { getSettings } from '../store/effects'
import { Spinner } from '@wordpress/components'
import { Container } from 'reactstrap'

export const StateContext = createContext(State)

/**
 * App Component
 *
 * @returns {JSX.Element}
 */
function App (): JSX.Element {
  const [state, setState] = useState(State)

  useEffect(() => {
    const storeSubscription = settingsStore.subscribe(() => setState({ ...state, ...settingsStore.getState() }))
    return storeSubscription
  })

  return (
        <StateContext.Provider value={state}>
            {state.api.settings === 'loading' && <Spinner/>}
            <Container>
                <Settings page={state.sidebar.activeSidebarMenu}/>
            </Container>
        </StateContext.Provider>
  )
}
/**
 *  SettingsApp Compoent
 *
 * @export
 * @returns {JSX.Element}
 */
export function SettingsApp (): JSX.Element {
  useEffect(() => {
    getSettings().catch(error => error)
  })

  return <App/>
}
