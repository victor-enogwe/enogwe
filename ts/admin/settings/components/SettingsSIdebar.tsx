import * as React from 'react'
import { useContext } from '@wordpress/element'
import { Dashicon } from '@wordpress/components'
import { settingsStore } from '../store'
import { ACTIVATE_SIDEBAR_MENU } from '../store/actions'
import { SidebarProps, Tab, GenericAction, MenuTabNames } from 'global'
import { StateContext } from './SettingsApp'
import { Nav, NavItem, NavLink } from 'reactstrap'

/**
 * SettingsSidebar Component
 *
 * @export
 * @param {SidebarProps} props
 * @returns {JSX.Element}
 */
export function SettingsSidebar (props: SidebarProps): JSX.Element {
    const maintenance = useContext(StateContext).maintenance
    const tabs: Tab[] = props.tabNames.map(name => ({ name, className: name, title: name, icon: props.tabIcons[name] }))
    const disabled = (tab: Tab) => tab.title === 'Maintenance' && !maintenance
    const isActive = (tab: Tab) => props.activeSidebarMenu === tab.title
    const activateMenu = (tab: Tab) => () => settingsStore.dispatch<GenericAction<MenuTabNames>>({
        type: ACTIVATE_SIDEBAR_MENU, data: tab.name
    })
    const Tabs = tabs.map((tab, index) => (
        <NavItem key={`settings_sidebar_${index}`}>
            <NavLink
                className={tab.className}
                onClick={activateMenu(tab)}
                label={tab.title}
                disabled={disabled(tab)}
                active={isActive(tab)}
            >
                <Dashicon icon={tab.icon}/> {tab.title}
            </NavLink>
        </NavItem>
    ))

    return <Nav pills={true} vertical={true} navbar={true}>{Tabs}</Nav>
}
