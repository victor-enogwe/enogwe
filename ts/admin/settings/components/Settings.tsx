import * as React from 'react'
import { Row, Col, CardHeader, CardTitle, CardBody } from 'reactstrap'
import classnames from 'classnames'
import { PropsWithChildren } from '@types/@wordpress/element'
import { __ } from '@types/@wordpress/i18n'
import { IconButton } from '@types/@wordpress/components'
import { SettingsSidebar } from './SettingsSIdebar'
import { tabNames, tabIcons } from '../constants'
import { SettingsPageProps } from 'global'
import { SettingsPage } from './SettingsPage'

/**
 * Settings Component
 *
 * @export
 * @param {PropsWithChildren<SettingsPageProps>} props
 * @returns {JSX.Element}
 */
export function Settings (props: PropsWithChildren<SettingsPageProps>): JSX.Element {
  return (
        <Row noGutters={true} className={classnames('flex-fill', 'flex-grow-1')}>
            <Col md="3" className={classnames('card', 'p-0', 'border-0', 'bg-light', 'd-none', 'd-md-block')}>
                <CardHeader>
                    <CardTitle className={classnames('d-flex', 'mb-0')}>
                        <IconButton
                            isLink={true}
                            // onClick={expandMenu}
                            isSmall={true}
                            icon="menu"
                            size={29}
                            className={classnames('border-0', 'shadow-none')}
                        />
                        {__('Theme Settings')}
                    </CardTitle>
                </CardHeader>
                <CardBody>
                    <SettingsSidebar tabNames={tabNames} tabIcons={tabIcons} activeSidebarMenu={props.page}/>
                </CardBody>
            </Col>
            <SettingsPage page={props.page}/>
        </Row>
  )
}
