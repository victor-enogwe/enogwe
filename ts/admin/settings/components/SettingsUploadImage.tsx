import * as React from 'react'
import { PropsWithChildren } from '@types/@wordpress/element'
import { __ } from '@types/@wordpress/i18n'
import { IconButton, Placeholder } from '@types/@wordpress/components'
import { MediaInfo, ImageProps } from 'global'
import { settingStyles } from '../styles/settings'
import { Card, Row, Col, CardBody } from 'reactstrap'
import classnames = require('classnames')

/**
 * SettingsUploadImage Component
 *
 * @export
 * @param {PropsWithChildren<ImageProps>} props
 * @returns {JSX.Element}
 */
export function SettingsUploadImage (props: PropsWithChildren<ImageProps>): JSX.Element {
  const { media, onDelete, onClick, label } = props
  const NoImage = () => <Placeholder icon="format-image" style={settingStyles.placeholder}/>
  const createSrcSet = (image: MediaInfo): string => Object.values(image.sizes).map(src => src.url).join(', ')

  return (
        <Card className={classnames('border-0', 'flex-fill', 'flex-grow-1', 'justify-content-center', 'align-items-center')}>
            <Row noGutters={true} className="w-md-75">
                <Col>{media ? <img srcSet={createSrcSet(media)}/> : <NoImage />}</Col>
                <Col>
                    <CardBody>
                        <IconButton isPrimary={true} icon="upload" onClick={onClick}>{__(`Upload ${label}`)}</IconButton>
                        <IconButton isDefault={true} disabled={!media} icon="trash" onClick={onDelete}>{__(`Delete ${label}`)}</IconButton>
                    </CardBody>
                </Col>
            </Row>
        </Card>
  )
}
