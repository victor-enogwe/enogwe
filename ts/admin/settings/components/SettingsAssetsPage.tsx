import * as React from 'react'
import { useContext } from '@wordpress/element'
import { PanelBody, Panel, PanelRow } from '@wordpress/components'
import { UPLOAD_LOGO, UPLOAD_FAVICON, UPLOAD_PICTURE, DELETE_FAVICON, DELETE_PICTURE, DELETE_LOGO } from '../store/actions'
import { StateContext } from './SettingsApp'
import { SettingsUploadImage } from './SettingsUploadImage'
import { settingStyles } from '../styles/settings'
import { handleMediaUpload, handleMediaDelete } from '../store/effects'

/**
 * SettingsAssetPage Componennt
 *
 * @export
 * @returns {JSX.Element}
 */
export function SettingsAssetsPage (): JSX.Element {
    const { assets } = useContext(StateContext)
    const logo = assets && assets.logo
    const favicon = assets && assets.favicon
    const picture = assets && assets.picture

    return (
        <div style={settingStyles.assetPage}>
            <Panel>
                <PanelBody initialOpen={true} title="LOGO">
                    <PanelRow>
                        <SettingsUploadImage
                            label="Logo"
                            onClick={handleMediaUpload(UPLOAD_LOGO)}
                            onDelete={handleMediaDelete(DELETE_LOGO)}
                            media={logo}
                        />
                    </PanelRow>
                </PanelBody>
            </Panel>
            <Panel>
                <PanelBody initialOpen={false} title="FAVICON">
                    <PanelRow>
                        <SettingsUploadImage
                            label="Favicon"
                            onClick={handleMediaUpload(UPLOAD_FAVICON)}
                            onDelete={handleMediaDelete(DELETE_FAVICON)}
                            media={favicon}
                        />
                    </PanelRow>
                </PanelBody>
            </Panel>
            <Panel>
                <PanelBody initialOpen={false} title="PROFILE PICTURE">
                    <PanelRow>
                        <SettingsUploadImage
                            label="Picture"
                            onClick={handleMediaUpload(UPLOAD_PICTURE)}
                            onDelete={handleMediaDelete(DELETE_PICTURE)}
                            media={picture}
                        />
                    </PanelRow>
                </PanelBody>
            </Panel>
        </div>
    )
}
