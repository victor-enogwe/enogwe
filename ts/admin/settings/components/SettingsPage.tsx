import * as React from 'react'
import { __ } from '@types/@wordpress/i18n'
import classnames from 'classnames'
import { PropsWithoutRef, useContext } from '@types/@wordpress/element'
import { Button, ToggleControl } from '@types/@wordpress/components'
import { SettingsPages, SettingsPageProps } from '@types/global'
import { SettingsLayoutPage } from './SettingsLayoutPage'
import { SettingsAssetsPage } from './SettingsAssetsPage'
import { StateContext } from './SettingsApp'
import { settingsStore } from '../store'
import { saveSettings, toggleMaintenance } from '../store/effects'
import { Col, CardHeader, CardBody, CardTitle } from 'reactstrap'

/**
 * SettingsPage Component
 *
 * @export
 * @param {PropsWithoutRef<SettingsPageProps>} props
 * @returns {JSX.Element}
 */
export function SettingsPage (props: PropsWithoutRef<SettingsPageProps>): JSX.Element {
  const { maintenance: { active }, api: { settings: loading } } = useContext(StateContext)
  const isBusy = loading === 'loading'
  const setState = () => saveSettings(settingsStore.getState())
  const pages: SettingsPages = {
    Assets: <SettingsAssetsPage />,
    Layouts:  <SettingsLayoutPage />
  }
  const Page = pages[props.page]

  return (
        <Col md="9" className={classnames('card', 'p-0', 'border-left-0')}>
            <CardHeader className={classnames('d-flex', 'justify-content-between', 'align-items-center')}>
                <CardTitle className={classnames('d-none', 'd-md-block', 'mb-n1')}>{props.page}</CardTitle>
                <div className="mb-n3">
                    <ToggleControl label={__('Maintenance', 'enogwe')} checked={active} onChange={toggleMaintenance}/>
                </div>
                <Button
                    isBusy={isBusy}
                    disabled={isBusy}
                    isPrimary={true}
                    label={__('Save  Settings', 'enogwe')}
                    onClick={setState}
                >
                    Save Settings
                </Button>
            </CardHeader>
            <CardBody>{Page}</CardBody>
        </Col>
  )
}
