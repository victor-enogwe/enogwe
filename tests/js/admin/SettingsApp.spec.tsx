import * as React from 'react'
import { shallow } from 'enzyme'
import { SettingsApp } from '../../../ts/admin/settings/components/SettingsApp'

describe('MyComponent', () => {
  it('should render correctly and have no props', () => {
    const component = shallow(<SettingsApp/>)
    expect(component.props()).toEqual({ })
    expect(component).toMatchSnapshot()
  })
})
