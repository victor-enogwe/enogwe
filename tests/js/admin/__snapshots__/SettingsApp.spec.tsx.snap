// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly and have no props 1`] = `<App />`;
